src/appleseed/renderer/meta/tests/test_projectfilewriter.cpp

```diff
@@ -287,4 +287,22 @@ TEST_SUITE(Renderer_Modeling_Project_ProjectFileWriter)
             string("curve_object.txt"),
             get_assembly()->objects().get_by_name("curve_object")->get_parameters().get("filepath"));
     }
+
+    TEST_CASE_F(Write_PackValidProject, Fixture)
+    {
+        const string PackedProjectName = "test_pack_valid_project.appleseedz";
+
+        create_project();
+
+        const bool success =
+            ProjectFileWriter::write(
+                m_project.ref(),
+                (m_output_directory / PackedProjectName).string().c_str(),
+                ProjectFileWriter::Defaults
+            );
+
+        ASSERT_TRUE(success);
+        ASSERT_TRUE(exists(m_output_directory / PackedProjectName));
+        ASSERT_FALSE(exists((m_output_directory / PackedProjectName).replace_extension(".unpacked")));
```

Comment:
Use the actual file/directory name in

    exists((m_output_directory / PackedProjectName).replace_extension(".unpacked"))

instead of using `replace_extension()`:

    exists(m_output_directory / "test_pack_valid_project.unpacked")

It's best to avoid as much logic as possible in unit tests, especially in assertions.